src/input.c

```diff
@@ -232,6 +232,22 @@ metadata_get(struct input_source *source)
   return NULL;
 }
 
+static struct timespec *
+ts_get(struct input_source *source)
+{
+  static struct timespec ts = {0, 0};
+  int ret;
+
+  if (!inputs[source->type]->ts_get)
+    return NULL;
+
+  ret = inputs[source->type]->ts_get(&ts, source);
+  if (ret < 0)
+    return NULL;
+
+  return &ts;
```

Comment:
ts is statically allocated so will go out of scope when the function exist. So this will return an invalid pointer. You could instead have the function as:

```
static int
ts_get(struct timespec *ts, struct input_source *source)
```
Then let the caller allocate ts. That way it is also the same as int (*ts_get).

Reply:
Thank you. That was a silly mistake of mine. I will fix it.

Reply:
It is my understanding that it was not a technically incorrect implementation. You are correct that the statically allocated ts goes out of scope when the function exits, but the pointer returned is not invalid. The variable ts is out of scope, but because it is statically allocated, the memory address is permanent and still contains valid data. 
However, I can certainly appreciate that it might be a style that is inconsistent with the rest of OwnTone, so no problem whatsoever to change it. I'm not great at github, so hopefully I can get the change I've made added to this PR.

Reply:
Sorry, your right, I didn't notice that the declaration of struct timespec had `static`. With that it will indeed remain valid. Having a static variable in a function is unusual, I think it's the first time I've seen it.